pkg/apis/autoscaling/validation/validation.go

```diff
@@ -183,6 +190,32 @@ func validateObjectSource(src *autoscaling.ObjectMetricSource, fldPath *field.Pa
 	return allErrs
 }
 
+func validateExternalSource(src *autoscaling.ExternalMetricSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(src.MetricName) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("metricName"), "must specify a metric name"))
```

Comment:
since this will be used as a single path segment in URL-building, this should be validated with `path.IsValidPathSegmentName`